src/fixed.ts

```diff
@@ -0,0 +1,144 @@
+import { MessageChannel, SHARE_ENV, Worker, isMainThread } from 'worker_threads'
+
+function empty () {}
+const _void = {}
+
+export type Draft<T> = { -readonly [P in keyof T]?: T[P] }
+
+export type WorkerWithMessageChannel = Worker & Draft<MessageChannel>
+
+export interface FixedThreadPoolOptions {
+  /**
+   * A function that will listen for error event on each worker thread.
+   */
+  errorHandler?: (this: Worker, e: Error) => void
+  /**
+   * A function that will listen for online event on each worker thread.
+   */
+  onlineHandler?: (this: Worker) => void
+  /**
+   * A function that will listen for exit event on each worker thread.
+   */
+  exitHandler?: (this: Worker, code: number) => void
+  /**
+   * This is just to avoid not useful warnings message, is used to set `maxListeners` on event emitters (workers are event emitters).
+   *
+   * @default 1000
+   */
+  maxTasks?: number
+}
+
+/**
+ * A thread pool with a static number of threads, is possible to execute tasks in sync or async mode as you prefer.
+ *
+ * This pool will select the worker thread in a round robin fashion.
+ *
+ * @author [Alessandro Pio Ardizio](https://github.com/pioardi)
+ * @since 0.0.1
+ */
+export default class FixedThreadPool<Data = any, Response = any> {
+  public readonly workers: WorkerWithMessageChannel[] = []
+  public nextWorker: number = 0
+
+  // threadId as key and an integer value
+  public readonly tasks: Map<WorkerWithMessageChannel, number> = new Map<
+    WorkerWithMessageChannel,
+    number
+  >()
+
+  protected _id: number = 0
+
+  /**
+   * @param numThreads Num of threads for this worker pool.
+   * @param filePath A file path with implementation of `ThreadWorker` class, relative path is fine.
+   * @param opts An object with possible options for example `errorHandler`, `onlineHandler`. Default: `{ maxTasks: 1000 }`
+   */
+  public constructor (
+    public readonly numThreads: number,
+    public readonly filePath: string,
+    public readonly opts: FixedThreadPoolOptions = { maxTasks: 1000 }
+  ) {
+    if (!isMainThread) {
+      throw new Error('Cannot start a thread pool from a worker thread !!!')
+    }
+    if (!this.filePath) {
+      throw new Error('Please specify a file with a worker implementation')
+    }
+
+    for (let i = 1; i <= this.numThreads; i++) {
+      this._newWorker()
+    }
+  }
+
+  public async destroy (): Promise<void> {
+    for (const worker of this.workers) {
+      await worker.terminate()
+    }
+  }
+
+  /**
+   * Execute the task specified into the constructor with the data parameter.
+   *
+   * @param data The input for the task specified.
+   * @returns Promise that is resolved when the task is done.
+   */
+  public async execute (data: Data): Promise<Response> {
```

Comment:
I didn't remove this `async` because I didn't want to change the code directly with this PR
But I think it can be removed because it's not necessary

This function just doesn't have an `await` in it

Reply:
But it returns a Promise so it's a valid syntax and does not hurt at all the transpilation. 
And some coding style rules enforce the async keyword before methods or functions returning Promise.